operator/src/main/java/oracle/kubernetes/weblogic/domain/model/DomainResource.java

```diff
@@ -893,7 +893,7 @@ class Validator {
     private final Set<String> serverNames = new HashSet<>();
 
     List<String> getValidationFailures(KubernetesResourceLookup kubernetesResources) {
-      getSimpleValidationFailures();
+      getFatalValidationFailures();
```

Comment:
I hadn't noticed this before. We have two methods that are populating the same collection, and we count on the fact that when we call `getFatalValidationFailures`, no other code has populated the `failures` collection. This seems like a problem waiting to happen. It would be much cleaner to have both `getFatalValidationFailures' and `getCrossReferenceValidationFailures` create collections, which they pass to the various validation methods, and then return that populated collection. 

In the short term, if we don't want to make that change, at the very least, we should rename both methods so that is clear that they simply populate the failure list, and create a separate `getFatalValidationFailures'  which calls the first one and returns the result.

Reply:
Good catch. working on a fix.

Reply:
Having looked closer, I don't think this is a problem.  Each method constructs a new Validator object, and therefore, populates different failures collections.

  public List<String> getFatalValidationFailures() {
    return new Validator().getFatalValidationFailures();
  }

  // used by the operator
  public List<String> getValidationFailures(KubernetesResourceLookup kubernetesResources) {
    return new Validator().getValidationFailures(kubernetesResources);
  }